pkg/object/pipeline/mock.go

```diff
@@ -199,3 +193,13 @@ func MockFilterSpec(super *supervisor.Supervisor, rawSpec map[string]interface{}
 		filterSpec: filterSpec,
 	}
 }
+
+// MockGetFilter is used to get running filter from pipeline
+func MockGetFilter(p *Pipeline, name string) Filter {
```

Comment:
Could this be getter with signature `func (p *Pipeline) Filter(name string)` ?

Reply:
this function is used for test only... i think pipeline maybe should not provide these kind of methods to get or change state of filter except for testing purpose...

Reply:
Alright, good point! If it's used for test only, maybe consider moving it to test file to avoid confusion..

Reply:
it used for test not in pipeline directory. but other file directory, like `MQTTProxy` use this method to get publish backend filter and check if packets were published to backend and maybe later we add more tests in HTTPServer and use this method to check if filter is in right status...